akka-actor/src/main/scala/akka/io/InetAddressDnsResolver.scala

```diff
@@ -29,41 +30,55 @@ class InetAddressDnsResolver(cache: SimpleDnsCache, config: Config) extends Acto
   // Deprecated JVM property key, keeping for legacy compatibility; replaced by NegativeCachePolicyProp
   private final val NegativeCachePolicyPropFallback = "sun.net.inetaddr.negative.ttl"
 
-  // default values (-1 and 0 are magic numbers, trust them)
-  private final val Forever = -1
-  private final val Never = 0
-  private final val DefaultPositive = 30
+  private final val DefaultPositive = Ttl.fromPositive(30.seconds)
 
-  private lazy val cachePolicy: Int = {
-    val n = Try(Security.getProperty(CachePolicyProp).toInt)
+  private lazy val defaultCachePolicy: CachePolicy =
+    Try(Security.getProperty(CachePolicyProp).toInt)
       .orElse(Try(System.getProperty(CachePolicyPropFallback).toInt))
+      .map(parsePolicy)
       .getOrElse(DefaultPositive) // default
```

Comment:
if these properties are misconfigured, e.g. not a number, then we are silently using the defaults? logging?

Reply:
Hmmm, good idea. 

The class currently doesn't have a logger injected so I'm tempted to add a good old `LoggerFactory.getLogger` but I'm not sure that's good idea in the Akka codebase, though.

Reply:
The alternative is a new constructor argument (`logger: LoggingAdapter`) but this is public APi (not even `@ApiMayChange`). :-(

Reply:
```
extends Actor with ActorLogging
```

then you have `log.`

Reply:
🤦‍♂️ ofc